src/workerd/io/io-context.h

```diff
@@ -726,6 +726,28 @@ class IoContext final: public kj::Refcounted, private kj::TaskSet::ErrorHandler
       const WarningAggregator::Key& key,
       kj::Function<kj::Own<WarningAggregator>(IoContext&)> load);
 
+  // The IncomingRequest that is currently considered "current". This is always the
+  // latest-starting request that hasn't yet completed.
+  //
+  // For stateless requests, there is only ever one IncomingRequest per IoContext. For
+  // actors, there is one IoContext per actor, and each incoming request to the actor
+  // creates a new  IncomingRequest.
+  //
+  // The current request is tracked for metrics, logging, and tracing purposes. Any resource
+  // usage on the part of the actor, including outgoing subrequests, is attributed to the current
+  // request for logging and tracing. This is a hack, we don't actually know which request
+  // "caused" any particular resource usage, so this is merely our best guess.
+  //
+  // The IoChannelFactory must also be accessed through the currentIncomingRequest because it has
+  // some tracing context built in.
+  IncomingRequest& getCurrentIncomingRequest() {
+    KJ_REQUIRE(!incomingRequests.empty(), "the IoContext has no current IncomingRequest");
+    return incomingRequests.front();
+  }
+  IoChannelFactory& getIoChannelFactory() {
```

Comment:
I'd prefer if we could keep this private, too -- I describe a possible way to do so in the counterpart internal PR. However, I can live with it being public if time is of the essence, so I'm approving.

Reply:
Added a todo for now